api4/status.go

```diff
@@ -99,7 +99,11 @@ func updateUserStatus(c *Context, w http.ResponseWriter, r *http.Request) {
 	case "away":
 		c.App.SetStatusAwayIfNeeded(c.Params.UserId, true)
 	case "dnd":
-		c.App.SetStatusDoNotDisturb(c.Params.UserId)
+		if c.Params.DNDEndTime != "" {
+			c.App.SetStatusDoNotDisturbTimed(c.Params.UserId, c.Params.DNDEndTime)
+		} else {
+			c.App.SetStatusDoNotDisturb(c.Params.UserId)
+		}
```

Comment:
I don't think there is a need for a separate function for this. We can pass the `c.Params.DNDEndTime` in `c.App.SetStatusDoNotDisturb` itself and do the conditional operation there.

Reply:
that will require changing signature of SetStatusDoNotDisturb which will break backward compatibility.

Reply:
The app layer is our internal code. It should have been an internal package anyways. It's fine to break backward compatibility.